### PR TITLE
Suppress DatabaseNotReady errors when reporting to senty

### PR DESCRIPTION
**Pull request by jhugman**
Fixes SDK-368.

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt [hunk 1]
```diff
@@ -393,11 +393,13 @@ open class Nimbus(
...
-            try {
-                errorReporter("Error in Nimbus Rust", e)
-            } catch (e1: Throwable) {
-                logger("Exception calling rust: $e")
-                logger("Exception reporting the exception: $e1")
+            if (e !is NimbusException.DatabaseNotReady) {
+                try {
+                    errorReporter("Error in Nimbus Rust", e)
+                } catch (e1: Throwable) {
+                    logger("Exception calling rust: $e")
```
Comment (tarikeshaq) [inline components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt:400]: I'm not sure (and this is not related to your change specifically, since you just moved it into the if statement) but should this be in the "try" block? since it's logging that the issue was in the rust?

**Review by tarikeshaq — APPROVED** [review]
Makes sense to me! Is this worth a changelog entry?

File: components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt (hunk 1 above)
Comment (jhugman) [reply components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt:400]: That's a great question. 

The inner try/catch is guarding ourselves from the `errorReporter` crashing. If it does we log the error with the logger, which we're betting on not failing. We also log the error in the `errorReporter`.
